proxy/contracts/Messages.sol

```diff
@@ -119,6 +121,17 @@ library Messages {
         uint256 tokenId;
     }
 
+    /**
+     * @dev Structure for describing base ERC721 with metadata.
+     */
+    struct TransferErc721MessageWithMetadata {
```

Comment:
`TransferErc721MessageWithMetadata` should be inherited from `TransferErc721Message` but not `BaseMessage`.

Reply:
Good point!